### PR TITLE
Separate Functional Tests

### PR DESCRIPTION
**Pull request by xavierchanth**
**- What I did**

Separated functional tests:

- Created composite workflows for:
  - setup job
  - generating a single application from at_app
- Independent dart analysis workflows for each package (to allow README badges)
- Generating and analyzing each generated app in parallel to reduce overall test duration

**- How I did it**

 - Local changes

**- How to verify it**

- Check the actions, verify against my changes

**- Description for the changelog**
Separate Functional Tests

### CODE REVIEW

File: .github/workflows/analyze_at_app.yaml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    branches:
+      - trunk
+
+jobs:
+  analyze_build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v3
+      - uses: ./.github/workflows/setup_job
```
Comment (cpswan) [inline .github/workflows/analyze_at_app.yaml]: This is my first time seeing this pattern of including a composite action, so it probably deserves a comment saying what's happening here.

(Initially I'm wondering how `dart analyze` can be used without installing Dart)?

We should probably be having a higher level conversation about establishing an actions repo with boilerplate we're likely to use across other repos.

File: .github/workflows/setup_job/action.yaml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+description: Setup job for at_app testing
+
+runs:
+  using: composite
+  steps:
+    - uses: subosito/flutter-action@v2
+      with:
+        channel: 'stable'
+    - run: flutter --version
+      shell: bash
```
Comment (cpswan) [inline .github/workflows/setup_job/action.yaml:11]: Does bash have to be specified each time, or is it the default?

**Review by cpswan — APPROVED** [review]

File: .github/workflows/analyze_at_app.yaml (hunk 1 above)
Comment (xavierchanth) [reply .github/workflows/analyze_at_app.yaml]: That's a good idea!

File: .github/workflows/setup_job/action.yaml (hunk 2 above)
Comment (xavierchanth) [reply .github/workflows/setup_job/action.yaml:11]: It is optional, but the linter wasn't detecting the correct schema definition unless I included it.

https://docs.github.com/en/actions/creating-actions/metadata-syntax-for-github-actions#runsstepsshell

File: .github/workflows/analyze_at_app.yaml (hunk 1 above)
Comment (xavierchanth) [reply .github/workflows/analyze_at_app.yaml]: Added the comment

**Pull request merged**